### PR TITLE
feat: add jms implementation 

### PR DESCRIPTION
**Pull request by jeskosda**

### CODE REVIEW

File: application/build.gradle [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+	implementation project(':domain')
 
-    developmentOnly 'org.springframework.boot:spring-boot-devtools'
-    testImplementation 'org.springframework.boot:spring-boot-starter-test'
-}
\ No newline at end of file
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	testImplementation 'org.awaitility:awaitility:4.2.0'
+	testImplementation 'org.springframework.cloud:spring-cloud-contract-wiremock'
+	testImplementation "org.junit.jupiter:junit-jupiter:5.8.1"
```
Comment (huebner-j) [inline application/build.gradle:16]: you need a version here?

File: application/build.gradle [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+	implementation project(':infrastructure-petstore-rest-client')
+	implementation project(':infrastructure-pet-jms')
+	implementation project(':domain')
 
-    developmentOnly 'org.springframework.boot:spring-boot-devtools'
-    testImplementation 'org.springframework.boot:spring-boot-starter-test'
-}
\ No newline at end of file
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	testImplementation 'org.awaitility:awaitility:4.2.0'
```
Comment (huebner-j) [inline application/build.gradle:14]: Can we centralise these versions? I guess, we need the libraries in other modules as well

File: application/build.gradle (hunk 1 above)
Comment (jeskosda) [reply application/build.gradle:16]: I dont need the dep at all,  i would guess :D

File: application/build.gradle (hunk 2 above)
Comment (jeskosda) [reply application/build.gradle:14]: We could provide it to all subprojects. But we can also wait until it really needed.

File: application/src/test/java/today/stepbeyond/examples/springbootexamples/application/domain/BirthOfDogIT.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+              .isPresent()
+              .get()
+              .isExactlyInstanceOf(Dog.class)
+              .extracting(Pet::getName)
+              .isEqualTo("Lilly");
+          softly
+              .assertThat(petRepository.find(newDog.getId()))
+              .isPresent()
+              .get()
+              .isNotInstanceOf(Cat.class);
```
Comment (huebner-j) [inline application/src/test/java/today/stepbeyond/examples/springbootexamples/application/domain/BirthOfDogIT.java:82]: NIT: I think there is something like `containsExactlyInstanceOf(Cat.class)` in assertj library

File: infrastructure-pet-jms/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/jms/pet/event/JmsPetEventProducer.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+public class JmsPetEventProducer implements PetEventPublisher {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JmsPetEventProducer.class);
+
+  private final JmsTemplate jmsTemplate;
+  private final String petIsBornDestination;
+
+  public JmsPetEventProducer(
+      JmsTemplate jmsTemplate,
+      @Value("${app.pet.events.pet.destination}") String petIsBornDestination) {
```
Comment (huebner-j) [inline infrastructure-pet-jms/src/main/java/today/stepbeyond/examples/springbootexamples/infrastructure/jms/pet/event/JmsPetEventProducer.java:23]:
```suggestion
      @Value("${app.pet.events.destination}") String petIsBornDestination) {
```

**Pull request merged**